File: app/src/main/java/com/kacper/and_krakgo/screens/sign_in/login/LoginContract.java
package com.kacper.and_krakgo.screens.sign_in.login;

/**
 * Created by kacper on 22/10/2017.
 */

public interface LoginContract {
    interface View{
        void onLoginSuccesfull();
        void showError(Exception exception);
    }
    interface Presenter{
        void loginWithEmail(String email, String password);
    }
}

File: app/src/main/java/com/kacper/krakgo/screens/sign_in/login/LoginContract.kt
package com.kacper.krakgo.screens.sign_in.login

import com.kacper.krakgo.mvp.MvpPresenter
import com.kacper.krakgo.mvp.MvpView

object LoginContract {
    interface View : MvpView{
        fun onLoginSuccesfull()
    }

    interface Presenter : MvpPresenter<View>{
        fun loginWithEmail(email: String, password: String)
    }
}